declaration/models.py

```diff
@@ -51,5 +47,30 @@ class Meta:
 
     def __str__(self):
         return f'User: {self.user} - tracking code: {self.tracking_code}'
+    
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.cache_status = self.status
+
+    def save(self, *args, **kwargs):
+        if self.pk is not None:
+            if self.status != self.cache_status:
+                status_history = PackageStatusHistory(package=self, status=self.status)
+                status_history.save()
 
+        self.cache_status = self.status
```

Comment:
You do not need this line at all. The next time django gets item from DB, it will update cache, as __init__ is called again